controllers/users.js

```diff
@@ -128,6 +128,20 @@ const getUsers = async (req, res) => {
       }
     }
 
+    if (transformedQuery?.filterBy === "overdue_tasks") {
+      try {
+        const users = await userQuery.getUserswithOverdueTasks(days);
+        return res.json({
+          message: "Users returned successfully!",
+          count: users.length,
+          users: users,
+        });
+      } catch (error) {
+        logger.error(`Error while fetching all users: ${error}`);
+        return res.boom.serverUnavailable("Something went wrong please contact admin");
+      }
+    }
```

Comment:
error handling logic seems appropriate. You catch errors that may occur during the execution of await userQuery.getUserswithOverdueTasks(days), log the error, and return a serverUnavailable response. Just ensure that the logger object is correctly configured in your application to log errors effectively.